core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -768,19 +769,19 @@ protected final void doStop() {
         threadPool.generic().execute(() -> {
             globalLock.writeLock().lock();
             try {
-                for (Iterator<NodeChannels> it = connectedNodes.values().iterator(); it.hasNext(); ) {
-                    NodeChannels nodeChannels = it.next();
-                    it.remove();
-                    IOUtils.closeWhileHandlingException(nodeChannels);
-                }
-
                 for (Map.Entry<String, List<Channel>> entry : serverChannels.entrySet()) {
                     try {
                         closeChannels(entry.getValue());
                     } catch (Exception e) {
                         logger.debug("Error closing serverChannel for profile [{}]", e, entry.getKey());
                     }
                 }
+
+                for (Iterator<NodeChannels> it = connectedNodes.values().iterator(); it.hasNext(); ) {
```

Comment:
why did you move this? can you leave a comment if it's important? also, we do it again in the finally close of the next statement. Did you find out why it needs to be done twice? if so, please add a comment :)